src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/ComparisonExpression.java

```diff
@@ -52,6 +53,13 @@ public ComparisonExpression(
    * @return {@link ComparisonExpression} with equal operator
    */
   public static ComparisonExpression eq(String path, Object value) {
+    if (!DocumentConstants.Fields.VALID_FIELD_NAME_PATTERN.matcher(path).matches()) {
```

Comment:
The constructor for ComparisionExpression is public and is not checked for valid pattern. We do support `$vector` as field name in filter with `$exists` operator.